library-valcon-internship/src/components/BookFormWrapper/BookFormWrapper.tsx

```diff
@@ -0,0 +1,278 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+
+import { useNavigate } from 'react-router-dom'
+
+import AuthorFormType from '../../models/AuthorFormType'
+import AuthorFormValidation from '../../models/AuthorFormValidation'
+import BookDetail from '../../models/BookDetail'
+import BookFormType from '../../models/BookFormType'
+import BookFormValidation from '../../models/BookFormValidation'
+import { addNewAuthor, getAllAuthors } from '../../services/AuthorService'
+import { addNewBook, editBook } from '../../services/BookService'
+import { convertAuthorsToAuthorDetails, convertBase64ToBlob } from '../../utils/Utils'
+import AuthorForm from '../AuthorForm/AuthorForm'
+import BookForm from '../BookForm/BookForm'
+import Modal from '../Modal/Modal'
+import './BookFormWrapper.css'
+
+interface BookFormWrapperProps {
+  closeModal?: () => void
+  book?: BookDetail
+  retrieveBook?: boolean
+  setRetrieveBook?: Dispatch<SetStateAction<boolean>>
+}
+
+const BookFormWrapper = ({ closeModal, book, retrieveBook,  setRetrieveBook }: BookFormWrapperProps) => {
+  const [ bookForm, setBookForm ] = useState<BookFormType>({
+    requestCover: book?.Cover ? '' : new Blob(),
+    cover: book?.Cover ? 'data:image/png;base64,' + book?.Cover : '',
+    title: book?.Title ? book?.Title : '',
+    description: book?.Description ? book?.Description : '',
+    isbn: book?.ISBN ? book?.ISBN : '',
+    quantity: book?.Quantity ? book?.Quantity.toString() : '',
+    releaseDate: book?.PublishDate ? new Date(book?.PublishDate) : null,
+    selectedAuthors: book?.Authors ? book?.Authors : [],
+    authors: []
+  })
```

Comment:
maybe it would be better to have initBookValues object with initial values. Then you will assign bookForm to a props.book or initBookValues, depending if props.book is passed. Later you could reuse initBookValues to reset form. Date you can convert immediately after you have received book from API.